dydx/dydxAnalytics/dydxAnalytics/AnalyticsEvent.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  AnalyticsEvent.swift
+//  dydxPresenters
+//
+//  Created by Rui Huang on 26/03/2024.
+//
+
+import Foundation
+import PlatformParticles
+import Utilities
+import FirebaseAnalytics
+
+//
+// Events defined in the v4-web repo.  Ideally, we should keep this in-sync with v4-web
+//
+// TODO: deprecate and replace with AnalyticsEventV2
+public enum AnalyticsEvent: String {
+    // App
+    case networkStatus = "NetworkStatus"
+
+    // Navigation
+    case navigatePage = "NavigatePage"
+    case navigateDialog = "NavigateDialog"
+    case navigateDialogClose = "NavigateDialogClose"
+    case navigateExternal = "NavigateExternal"
+
+    // Wallet
+    case connectWallet = "ConnectWallet"
+    case disconnectWallet = "DisconnectWallet"
+
+    // Onboarding
+    case onboardingStepChanged = "OnboardingStepChanged"
+    case onboardingAccountDerived = "OnboardingAccountDerived"
+    case onboardingWalletIsNonDeterministic = "OnboardingWalletIsNonDeterministic"
+
+    // Transfers
+    case transferFaucet = "TransferFaucet"
+    case transferFaucetConfirmed = "TransferFaucetConfirmed"
+    case transferDeposit = "TransferDeposit"
+    case transferWithdraw = "TransferWithdraw"
+
+    // Trading
+    case tradeOrderTypeSelected = "TradeOrderTypeSelected"
+    case tradePlaceOrder = "TradePlaceOrder"
+    case tradePlaceOrderConfirmed = "TradePlaceOrderConfirmed"
+    case tradeCancelOrder = "TradeCancelOrder"
+    case tradeCancelOrderConfirmed = "TradeCancelOrderConfirmed"
+
+    // Notification
+    case notificationAction = "NotificationAction"
+}
+
+public extension AnalyticsEventV2 {
+    enum OnboardingStep: String {
+        case chooseWallet = "ChooseWallet"
+        case keyDerivation = "KeyDerivation"
+        case acknowledgeTerms = "AcknowledgeTerms"
+        case depositFunds = "DepositFunds"
+    }
+}
+
+public extension AnalyticsEventV2 {
+    enum OnboardingState: String {
+        case disconnected = "Disconnected"
+        case walletConnected = "WalletConnected"
+        case accountConnected = "AccountConnected"
+    }
+}
+
+public enum AnalyticsEventV2: TrackableEvent, CustomDebugStringConvertible {
+    // TODO: add and replace all events
+    case appStart
+    case navigatePage(screen: ScreenIdentifiable)
+    case deepLinkHandled(url: String, succeeded: Bool)
+    case notificationPermissionsChanged(isAuthorized: Bool)
+    case onboardingStepChanged(step: OnboardingStep, state: OnboardingState)
+
+    public var name: String {
```

Comment:
The long switch statement can be messy when more events are added.  Maybe it's cleaner to define each event as a separate struct, e.g.,

```
struct NavigatePageEvent:  TrackableEvent {
  let screen: ScreenIdentifiable
  
  override var name: String ...
  
}
```